components/ws-manager-bridge/src/bridge.ts

```diff
@@ -36,6 +36,22 @@ import { WorkspaceInstanceController } from "./workspace-instance-controller";
 import { PrebuildUpdater } from "./prebuild-updater";
 import { RedisPublisher } from "@gitpod/gitpod-db/lib";
 
+/**
+ * TODO(se) use a proper distributed lock here
+ * This only works for a single instance of bridge, but that's fine as we run only one instance of ws-manager-bridge
+ * Ideally we'd use a redis-based distributed lock here, but that's not available in this component (only server)
+ */
+class PoorMensSyncer {
+    private readonly queues = new Map<string, Queue>();
+
+    public sync<T>(key: string, cb: () => Promise<T>): Promise<T> {
```

Comment:
I'm lagging context atm, but: What's the difference to the "queueing" by instanceId we do [here](https://github.com/gitpod-io/gitpod/pull/19124/files#diff-5c377b2cad37712ee6f512a499f1069f7641af5aab7dceaa7e8d033fc23d4780R178)? :thinking:

Reply:
Thanks, I didn't see that code. I'm still somewhat confident we overwrite state somewhere. But this looks indeed like it does exactly the same. So will close this PR and check whether this is caused by async updates in server.

Reply:
>  I'm still somewhat confident we overwrite state somewhere

:+1: Might very well be! Also, can still be an issue with that code - I did not do a careful review here, just wanted to point out what's there.